daemon/delete.go

```diff
@@ -102,8 +102,10 @@ func (daemon *Daemon) cleanupContainer(container *container.Container, forceRemo
 	// Save container state to disk. So that if error happens before
 	// container meta file got removed from disk, then a restart of
 	// docker should not make a dead container alive.
-	if err := container.ToDiskLocking(); err != nil {
-		logrus.Errorf("Error saving dying container to disk: %v", err)
+	if _, err := os.Stat(container.Root); err == nil {
```

Comment:
@mlaventure 
So what are you suggesting ? If I move that check inside `GetRootResourcePath()` and `container.Root` is not present, what path will it return ? 
We also don't want to do the rest of the operations in `ToDisk` function. So I believe the check is at the right place.